### ISSUE TITLE
[action] focus state and keyboard navigation in safari is not working

### ISSUE THREAD

**Issue by DintaMel** [Esri/calcite-components#14593]
**monday.com sync:** #12231757739

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Which Component

Action

### Actual Behavior

When I select an action in Safari, it does not have a focus state. It also can't be navigated via keyboard.

### Expected Behavior

When I select an action in Safari, it should have a focus state and be navigable via keyboard. 

Focus state in Chrome: 
<img width="233" height="67" alt="Image" src="https://github.com/user-attachments/assets/ae072fc0-c2c0-4467-9998-946c46f9e020" />

### Reproduction Sample

https://codepen.io/DintaMel-the-lessful/pen/jEybOGO?editors=100

### Reproduction Steps

1. Open [codepen](https://codepen.io/DintaMel-the-lessful/pen/jEybOGO?editors=100) in Safari
2. Select an action
3. Observe that the action doesn't have a focus state.
4. Try to access via keyboard.
5. Observe that it can't be access via keyboard.

### Reproduction Version

5.1.0

### Regression?

3.2.1

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Maps SDK for JavaScript

**Comment by annelfitz** [comment]
@DintaMel It's more than just the focus state displayed -- these actions are not keyboard accessible at all in Safari. I cannot navigate to them using the keyboard.

**Comment by DintaMel** [comment]
> [@DintaMel](https://github.com/DintaMel) It's more than just the focus state displayed -- these actions are not keyboard accessible at all in Safari. I cannot navigate to them using the keyboard.

@annelfitz Adjusted to include keyboard navigation.

**Comment by driskull** [comment]
This seems to just be safari quirks:

Safari doesn't focus buttons when clicked
https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/button#clicking_and_focus

Enable the safari option to tab to buttons
https://support.apple.com/en-is/guide/safari/ibrw1075/mac#:~:text=Press%20Tab%20to%20highlight%20each%20item%20on%20a%20webpage

**Comment by DintaMel** [comment]
> This seems to just be safari quirks:
> 
> Safari doesn't focus buttons when clicked https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/button#clicking_and_focus
> 
> Enable the safari option to tab to buttons https://support.apple.com/en-is/guide/safari/ibrw1075/mac#:~:text=Press%20Tab%20to%20highlight%20each%20item%20on%20a%20webpage

Interesting. I verified that it works with the Safari option to tab to buttons. 

@driskull is there a reason this works out of the box in version 3.2 without the Safari setting enabled?

**Comment by jcfranco** [comment]
It could be because we were [setting `tabindex` at 3.2](https://github.com/Esri/calcite-design-system/blob/%40esri/calcite-components%403.2.0/packages/calcite-components/src/components/action/action.tsx#L299) for consistent interaction (e.g., `disabled`) behavior as we didn't have access to `inert` at the time. It was removed in https://github.com/Esri/calcite-design-system/pull/12139 for a fix.

**Comment by geospatialem** [comment]
Chatted with the team offline and we are closing the above as out of scope with the lack of support with Safari. We could add internal documentation on how to [enable the Safari option](https://support.apple.com/en-is/guide/safari/ibrw1075/mac#:~:text=Press%20Tab%20to%20highlight%20each%20item%20on%20a%20webpage) for teams, since so many of our internal teams use Safari, but if so, this seems like a low priority for us to explore since its specific to browser and OS settings.

While we did have functionality behaving as-expected with `3.2` [other changes were performed](https://github.com/Esri/calcite-design-system/issues/14593#issuecomment-4772069190) that may have impacted the behavior, but its not something we'd like to revert or explore alternative considerations.

Closing the above out as unplanned.
